spec/features/emails_spec.rb

```diff
@@ -423,52 +422,43 @@
       expect(email).to have_body_text(investment.unfeasibility_explanation)
     end
 
-    scenario "Selected investment" do
+    scenario 'Selected investment' do
       author1 = create(:user)
       author2 = create(:user)
-      author3 = create(:user)
 
       investment1 = create(:budget_investment, :selected,   author: author1, budget: budget)
-      investment2 = create(:budget_investment, :selected,   author: author2, budget: budget)
-      investment3 = create(:budget_investment, :unselected, author: author3, budget: budget)
+      investment2 = create(:budget_investment, :unselected, author: author2, budget: budget)
 
       reset_mailer
       budget.email_selected
 
       expect(find_email(investment1.author.email)).to be
-      expect(find_email(investment2.author.email)).to be
-      expect(find_email(investment3.author.email)).not_to be
+      expect(find_email(investment2.author.email)).not_to be
 
       email = open_last_email
-      investment = investment2
-      expect(email).to have_subject("Your investment project '#{investment.code}' has been selected")
-      expect(email).to deliver_to(investment.author.email)
-      expect(email).to have_body_text(investment.title)
+      expect(email).to have_subject("Your investment project '#{investment1.code}' has been selected")
+      expect(email).to deliver_to(investment1.author.email)
+      expect(email).to have_body_text(investment1.title)
     end
 
-    scenario "Unselected investment" do
+    scenario 'Unselected investment' do
       author1 = create(:user)
       author2 = create(:user)
-      author3 = create(:user)
 
       investment1 = create(:budget_investment, :unselected, author: author1, budget: budget)
-      investment2 = create(:budget_investment, :unselected, author: author2, budget: budget)
-      investment3 = create(:budget_investment, :selected,   author: author3, budget: budget)
+      investment2 = create(:budget_investment, :selected,   author: author2, budget: budget)
 
       reset_mailer
       budget.email_unselected
 
       expect(find_email(investment1.author.email)).to be
-      expect(find_email(investment2.author.email)).to be
-      expect(find_email(investment3.author.email)).not_to be
+      expect(find_email(investment2.author.email)).not_to be
 
       email = open_last_email
-      investment = investment2
-      expect(email).to have_subject("Your investment project '#{investment.code}' has not been selected")
-      expect(email).to deliver_to(investment.author.email)
-      expect(email).to have_body_text(investment.title)
+      expect(email).to have_subject("Your investment project '#{investment1.code}' has not been selected")
+      expect(email).to deliver_to(investment1.author.email)
```

Comment:
In both examples expectations where done over `investment` and it was `investment = investment2`. Now directly its using `investment1` instead of `investment2`, why?

Reply:
Because such variable assignation was unnecessary, in my opinion. We were allocating more memory than necessary just for aesthetics.

Said variable assignation would be useful on scenarios where we wanted to compare an object's state before and after executing an action but since this is not the case, we can dispose the variable altogether without problems.

Reply:
I'm okey with the removal of `investment = investmentX` line. I'm concerned about why before the code expected investment2 author email (through `investment.author.email`, and now it's expecting investment1 author email (through investment1.author.email) Or in other words... why switch Investment**2** for Investment**1**... are those really interchangeable? or is this spec broken and no matter which investment we put there it will go ✅ ?

Reply:
Oh sorry, I misunderstood your previous comment.

I replaced `investment2` for `investment1` as I tend to use the first factory I create for the assertions at the end for said examples. `investment1`, for the `Selected investment` scenario, is the **selected** investment and for the `Unselected investment` scenario, `investment1` is the **unselected** investment

Reply:
After giving a read at the original file, I think there is no need for this changes as the objective of having 3 investments is to correctly assert that `Budget#email_selected` https://github.com/ayuntamientomadrid/consul/blob/master/app/models/budget.rb#L155-L159 works as expected by sending emails for all selected/unselected ones. Maybe the fact that its only checking the second email with `open_last_email` instead of checking both sent emails (for both selected or unselected investments) is what made you thought it was unnecessary data.